codegen/service/service_data.go

```diff
@@ -816,6 +816,40 @@ func buildMethodData(m *expr.MethodExpr, svcPkgName string, service *expr.Servic
 	return data
 }
 
+// mustInitPayload returns true if method Payload must be initialized.
+func mustInitPayload(m *expr.MethodExpr) bool {
+	if m.Payload.Type == expr.Empty {
+		return false
+	}
+
+	var httpMet *expr.HTTPEndpointExpr
+	if httpSvc := expr.Root.HTTPService(m.Service.Name); httpSvc != nil {
+		httpMet = httpSvc.Endpoint(m.Name)
+		if httpMet.SkipRequestBodyEncodeDecode && httpMet.HasBodyOnly() {
+			return false
+		}
+	}
+	return true
+}
+
+// mustInitResult returns true if method Result must be initialized.
+func mustInitResult(m *expr.MethodExpr) bool {
+	if m.Result.Type == expr.Empty {
+		return false
+	}
+
+	var httpMet *expr.HTTPEndpointExpr
+	if httpSvc := expr.Root.HTTPService(m.Service.Name); httpSvc != nil {
+		httpMet = httpSvc.Endpoint(m.Name)
+		for _, r := range httpMet.Responses {
+			if httpMet.SkipResponseBodyEncodeDecode && r.HasBodyOnly() {
+				return false
```

Comment:
Should this function only return `false` if *all* responses only define a body?

Reply:
With SkipResponseXXX there can be only be one successful response. Otherwise validation would fail https://github.com/goadesign/goa/blob/v3/expr/http_endpoint.go#L426-L428